dascore/proc/whiten.py

```diff
@@ -0,0 +1,225 @@
+"""Spectral whitening."""
+from __future__ import annotations
+
+import math
+
+import numpy as np
+import numpy.fft as nft
+import pandas as pd
+from scipy.ndimage import convolve1d
+from scipy.signal.windows import tukey
+
+from dascore.constants import PatchType
+from dascore.exceptions import ParameterError
+from dascore.utils.misc import check_filter_kwargs, check_filter_range
+from dascore.utils.patch import get_dim_sampling_rate
+
+
+def whiten(
+    patch: PatchType,
+    freq_smooth_size: float | None = None,
```

Comment:
Just "smooth_size" now since whitening can happen in wavenumber or other domains?

Reply:
Yes, done.